Tests/WhisperKitTests/MLXTests.swift

```diff
@@ -8,15 +7,21 @@ import XCTest
 
 final class MLXTests: XCTestCase {
 
-    override func setUpWithError() throws {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
-    }
-
-    override func tearDownWithError() throws {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
-    }
+    // MARK: - Feature Extractor Tests
 
-    func testExample() throws {
-        XCTAssertNotNil(MLXAudioEncoder())
+    func testLogmelOutput() async throws {
+        let audioSamples = [Float](repeating: 0.0, count: 16000)
+        let paddedSamples = try XCTUnwrap(
+            AudioProcessor.padOrTrimAudio(fromArray: audioSamples, startAt: 0, toLength: 480_000),
+            "Failed to pad audio samples"
+        )
+        let featureExtractor = MLXFeatureExtractor()
+        let melSpectrogram = try await XCTUnwrapAsync(
+            await featureExtractor.logMelSpectrogram(fromAudio: paddedSamples),
+            "Failed to produce Mel spectrogram from audio samples"
+        )
+        let expectedShape: [NSNumber] = [3000, 80]
```

Comment:
the output shape of the MLX feature extractor is different from the CoreML one

Reply:
Yes, the existing Core ML Melspectrogram prepares its output for direct consumption by the Core ML AudioEncoder which anticipates `[1, {80,128}, 1, 3000]` (N=128 for large variants). We need to standardize the interface such that Core ML and MLX blocks are fungible